### PR TITLE
build binaries with github actions

### PR DESCRIPTION
**Pull request by oxr463**
Closes: https://github.com/cs01/gdbgui/issues/331

### CODE REVIEW

**Comment by oxr463** [comment]
See also: https://rtyley.github.io/bfg-repo-cleaner

```sh
git clone --mirror https://github.com/cs01/gdbgui.git
bfg --delete-files downloads gdbgui.git
```

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - name: Compile binaries
+      run: |
+        nox -s docker_executables # linux, windows
+        nox -s build_executable_current_os # macos
+#    - name: Publish GitHub Release
+#      uses: fnkr/github-action-ghr@v1
+#      if: startsWith(github.ref, 'refs/tags/')
+#      env:
+#        GHR_PATH: FIXME
+#        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (bjorn3) [inline .github/workflows/release.yml:31]: You could also use https://github.com/actions/upload-artifact to attach artifacts to the current github actions run.

**Comment by cs01** [comment]
There is some documentation that points to those binaries which will need to be updated.

**Comment by oxr463** [comment]
Depends on: https://github.com/cs01/gdbgui/issues/335

**Comment by cs01** [comment]
Thanks, I will try to take a look this weekend

**Comment by oxr463** [comment]
> Thanks, I will try to take a look this weekend

Cool, once we have the binaries compiled, then we'll be good to go!

**Comment by cs01** [comment]
I made a new branch based off this and got everything working except for associating the artifacts with the release itself: https://github.com/cs01/gdbgui/tree/cs01/test-actions. I recommend resetting this PR to that branch.

Re associated artifacts with github releases: I can look up how to do it, but if you have any tips or can reset this PR to the `cs01/test-actions` branch, and make changes on top of that, that would be preferred. https://github.com/cs01/gdbgui/actions/runs/166916337 shows all the OS's have the executables built and artifacts stored. I don't think it's necessary to store artifacts for *all* pushes, only for actual github releases.

**Comment by cs01** [comment]
I will probably end up migrating all the Travis tests to GitHub actions. The actions will build the executable as well, and at the end the executable will be published.

**Comment by cs01** [comment]
Since this PR allowed edits from maintainers I went ahead and pushed to it. Thanks for kicking off this PR and motivating me to look into GitHub actions.

I'm going to land now with the following:
* Executables built on all pushes/pull requests for all OS's
* Artifacts are stored if on the master branch

Future work:
* Delete executables from source code. I am going to wait for a successful release with artifacts attached to a release in GitHub before doing this.
* Potentially automatically associate a build with a release. For now I'll just download the artifacts, then drag & drop them manually to associate with a release.

**Pull request merged**